### DISCUSSION TITLE
Updating Query Params with shallow causes page re-render / refresh, scroll to top effect

### DISCUSSION THREAD

**Discussion by Dellybro** [vercel/next.js#16736] [Help]
# Bug report

## Describe the bug

When updating the query params of a route using the following code 

```
router.push(`/jobs?${query}`, `/jobs?${query}`, {
          shallow: true,
});
```

The page scrolls to the top, and causes a "refresh/rerender"

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a page which is longer then your screen
2. Update query params progammatically

## Expected behavior

I would expect the page not to rerender, stay in the current scroll spot and update the query params without any extra things going on

## System information

- OS: macOs
- Chrome
- Version of Next.js: 9.3.4
- Version of Node.js: 12.x

**Comment by balazsorban44** [comment]
Take a look at the docs: https://nextjs.org/docs/routing/shallow-routing

By definition of the `useRouter` behaviour, there has to be a re-render to update the `router.query` object. `shallow` will only disable any data fetching mechanisms like `getStaticProps` or `getInitialProps`

**Comment by jamesmosier** [comment]
[Per the docs](https://nextjs.org/docs/routing/shallow-routing), your shallow routing code would need to look like:

```js
router.push(`/jobs?${query}`, undefined, {
  shallow: true,
});
```

**Reply by sanath-git** [reply]
@jamesmosier This works. Thanks :)
